pkg/k8shandler/clusterrestart.go

```diff
@@ -0,0 +1,552 @@
+				logrus.Errorf("unable to update node status: E: %s", err)
+			}
+		}
+	}
+
+	restarter.setRestarterClusterConditions(statusUpdate)
+	restarter.clusterStatus = &elasticsearchRequest.cluster.Status
+	return restarter.restartCluster()
+}
+
+func (elasticsearchRequest *ElasticsearchRequest) PerformFullClusterRestart(nodes []NodeTypeInterface) error {
+
+	r := ClusterRestart{
+		client:           elasticsearchRequest.esClient,
+		clusterName:      elasticsearchRequest.cluster.Name,
+		clusterNamespace: elasticsearchRequest.cluster.Namespace,
+		scheduledNodes:   nodes,
+	}
+
+	restarter := Restarter{
+		scheduledNodes:   nodes,
+		clusterName:      elasticsearchRequest.cluster.Name,
+		clusterNamespace: elasticsearchRequest.cluster.Namespace,
+		precheck:         r.ensureClusterHealthValid,
+		prep:             r.optionalSetPrimariesShardsAndFlush,
+		main:             elasticsearchRequest.scaleDownThenUpFunc(r),
+		post:             r.waitAllNodesRejoinAndSetAllShards,
+		recovery:         r.ensureClusterHealthValid,
+	}
+
+	statusUpdate := func() {
+		for _, node := range r.scheduledNodes {
+			nodeStatus := elasticsearchRequest.getNodeState(node)
+			nodeStatus.UpgradeStatus.ScheduledForCertRedeploy = v1.ConditionFalse
+
+			if err := elasticsearchRequest.setNodeStatus(node, nodeStatus, &elasticsearchRequest.cluster.Status); err != nil {
+				logrus.Errorf("unable to update node status: E: %s", err)
+			}
+		}
+	}
+
+	restarter.setRestarterClusterConditions(statusUpdate)
+	restarter.clusterStatus = &elasticsearchRequest.cluster.Status
+	return restarter.restartCluster()
+}
+
+func (elasticsearchRequest *ElasticsearchRequest) PerformNodeRestart(node NodeTypeInterface) error {
+
+	scheduledNode := []NodeTypeInterface{node}
+
+	r := ClusterRestart{
+		client:           elasticsearchRequest.esClient,
+		clusterName:      elasticsearchRequest.cluster.Name,
+		clusterNamespace: elasticsearchRequest.cluster.Namespace,
+		scheduledNodes:   scheduledNode,
+	}
+
+	restarter := Restarter{
+		scheduledNodes:   scheduledNode,
+		clusterName:      elasticsearchRequest.cluster.Name,
+		clusterNamespace: elasticsearchRequest.cluster.Namespace,
+		precheck:         r.ensureClusterHealthValid,
+		prep:             r.optionalSetPrimariesShardsAndFlush,
+		main:             elasticsearchRequest.scaleDownThenUpFunc(r),
+		post:             r.waitAllNodesRejoinAndSetAllShards,
+		recovery:         r.ensureClusterHealthValid,
+	}
+
+	statusUpdate := func() {
+		if err := elasticsearchRequest.setNodeStatus(node, restarter.nodeStatus, &elasticsearchRequest.cluster.Status); err != nil {
+			logrus.Errorf("unable to update node status: E: %s", err)
+		}
+	}
+
+	restarter.setRestarterNodeConditions(statusUpdate)
+
+	restarter.nodeStatus = elasticsearchRequest.getNodeState(node)
+	return restarter.restartCluster()
+}
+
+func (elasticsearchRequest *ElasticsearchRequest) PerformNodeUpdate(node NodeTypeInterface) error {
```

Comment:
This a single node update. Is this used anywhere else except `PerformRollingUpdate`?

Reply:
yes, it is called in cluster.go to resume an update